crypto/keyring_message.go

```diff
@@ -118,6 +118,17 @@ func (keyRing *KeyRing) VerifyDetachedEncrypted(message *PlainMessage, encrypted
 	return keyRing.VerifyDetached(message, signature, verifyTime)
 }
 
+// GetVerifiedSignatureTimestamp verifies a PlainMessage with a detached PGPSignature
+// returns the creation time of the signature if it succeeds
+// and returns a SignatureVerificationError if fails.
+func (keyRing *KeyRing) GetVerifiedSignatureTimestamp(message *PlainMessage, signature *PGPSignature, verifyTime int64) (int64, error) {
+	err := keyRing.VerifyDetached(message, signature, verifyTime)
```

Comment:
Here it verifies all the signatures, but then later we return the timestamp of the first one if any of them succeeds. We should return the timestamp of the succeeding one

Reply:
Can you have several sigs in a detached signature ?
If yes, how would I find out which one succeeded ?

Reply:
Yeah, it can happen. A simple solution here is to accept just one, since the crypto API `CheckDetachedSignatureAndHash` returns only the signer entity, and not the signature packet.

So basically instead of using `signature.getCreationTime()` you can just check if the first packet is a signature, then serialise just that one and verify it with `verifySignature`

Reply:
Wouldn't this lead to inconsistency between VerifyDetached and GetVerifiedSignatureTimestamp ?
Some sigs would be accepted by the first but not the other one

Reply:
Indeed. The other solution I see is to do the same but manually iterating over the signature packets, verifying one by one and returning the verifying one's timestamp

Reply:
Ok I tried going over each packet in a loop and return the time stamp for the first packet that verifies